### PR TITLE
[New Feature] Added New Command-line Utility

### PR DESCRIPTION
**Pull request by caneroj1**
Added a new command-line feature to Revel called "add." 

The command makes it easy to automatically create new controllers for your app, as it is instantiates them from a template file, and it also creates the views directory for that controller.

The command looks like `revel add [app_name] [controller_name]`.
For example, given a Revel application called testapp, `revel add testapp user` would create a new controller called user.go under /controllers and the User directory under /views.

### CODE REVIEW

**Comment by brendensoares** [comment]
I like this a lot, but I wonder if there is a more flexible way to allow code generation in the revel cmd tool?

Perhaps we will add this for now and brainstorm further... @revel/core thoughts?

**Comment by caneroj1** [comment]
Do you mean something that can generate code for models and other kinds of files instead of just controllers? That could definitely be useful.

**Comment by brendensoares** [comment]
@caneroj1 sorry for the long delay in response.

Would you be able to help @revel/core by creating a step by step guide on how to use this new command? I'm sure we could figure it out on our own, but we have a lot to do already and your help would be greatly appreciated!

File: revel/add.go [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+	fmt.Println("Done!")
+}
+
+func createApplicationPath(appName string) {
+	appPath = filepath.Join(srcRoot, filepath.FromSlash(appName))
+}
+
+func checkApplicationPath() {
+	fmt.Printf("Adding controller: %s.go\n", controller)
```
Comment (ptman) [inline revel/add.go]: I think this is a bit misleading. It claims to be adding a controller but in fact is checking the application path.

File: revel/add.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+	// now check if we can access the app directory
+	// under the app path
+	_, err = os.Stat(appPath + "/app")
+	if os.IsNotExist(err) {
+		errorf("Abort: Could not access the /app directory: %s\n", appPath+"/app")
+	}
+}
+
+func checkControllersDirectory() {
```
Comment (ptman) [inline revel/add.go]: Check ends up creating. Maybe ensure instead of check?

**Comment by ptman** [comment]
The basic approach for templating should still work if the conflicts are solved. But IMHO it doesn't give us much. Creating one file and one directory isn't that big of a task. What else should it do in order to be more useful? Routes?

**Comment by pedromorgan** [comment]
Thats probably what wew all after +1

**Comment by jeevatkm** [comment]
@ptman Thank you for time and PR review. 

So what I understand is, we take this PR as inspirational to come up with `create` or `generate` command in future with controller, routes creation etc.

**Comment by jeevatkm** [comment]
In-favour of #48 I'm cloning this one.

**Pull request closed without merging**